domain/removal/state/relation.go

```diff
@@ -0,0 +1,301 @@
+		return errors.Errorf("preparing relation removal: %w", err)
+	}
+
+	return errors.Capture(db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, stmt, removalRec).Run()
+		if err != nil {
+			return errors.Errorf("scheduling relation removal: %w", err)
+		}
+		return nil
+	}))
+}
+
+// GetRelationLife returns the life of the relation with the input UUID.
+func (st *State) GetRelationLife(ctx context.Context, rUUID string) (life.Life, error) {
+	db, err := st.DB()
+	if err != nil {
+		return -1, errors.Capture(err)
+	}
+
+	var relationLife entityLife
+	relationUUID := entityUUID{UUID: rUUID}
+
+	stmt, err := st.Prepare(`
+SELECT &entityLife.life_id
+FROM   relation
+WHERE  uuid = $entityUUID.uuid`, relationLife, relationUUID)
+	if err != nil {
+		return -1, errors.Errorf("preparing relation life query: %w", err)
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, stmt, relationUUID).Get(&relationLife)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return relationerrors.RelationNotFound
+		} else if err != nil {
+			return errors.Errorf("running relation life query: %w", err)
+		}
+
+		return nil
+	})
+
+	return relationLife.Life, errors.Capture(err)
+}
+
+// UnitNamesInScope returns the names of units in
+// the scope of the relation with the input UUID.
+func (st *State) UnitNamesInScope(ctx context.Context, rUUID string) ([]string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	type uName struct {
+		Name string `db:"name"`
+	}
+
+	relationUUID := entityUUID{UUID: rUUID}
+
+	stmt, err := st.Prepare(`
+SELECT &uName.name 
+FROM   relation_endpoint re 
+       JOIN relation_unit ru ON re.uuid = ru.relation_endpoint_uuid
+       JOIN unit u on ru.unit_uuid = u.uuid 
+WHERE  re.relation_uuid = $entityUUID.uuid`, uName{}, relationUUID)
+	if err != nil {
+		return nil, errors.Errorf("preparing relation scopes query: %w", err)
+	}
+
+	var inScope []uName
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, stmt, relationUUID).GetAll(&inScope)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return nil
+		} else if err != nil {
+			return errors.Errorf("running relation scopes query: %w", err)
+		}
+
+		return nil
+	})
+
+	return transform.Slice(inScope, func(n uName) string { return n.Name }), errors.Capture(err)
+}
+
+// DeleteRelationUnits deletes all relation unit records and their
+// associated settings for a relation. It effectively departs all
+// units from the scope of the input relation immediately.
+func (st *State) DeleteRelationUnits(ctx context.Context, rUUID string) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	relationUUID := entityUUID{UUID: rUUID}
+
+	settingsStmt, err := st.Prepare(`
+WITH rru AS (
+    SELECT ru.uuid, re.relation_uuid
+    FROM   relation_unit ru 
+           JOIN relation_endpoint re ON ru.relation_endpoint_uuid = re.uuid
```

Comment:
```suggestion
    FROM   relation_unit ru 
    JOIN   relation_endpoint re ON ru.relation_endpoint_uuid = re.uuid
```

Reply:
This is deliberate and quite common. It keeps a single indent associated with the the `FROM` stanza.